core-java/src/main/java/com/baeldung/bufferedreader/BufferedReaderExample.java

```diff
@@ -0,0 +1,108 @@
+package com.baeldung.bufferedreader;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class BufferedReaderExample {
+
+    public String readAllLines(BufferedReader reader) throws IOException {
+        StringBuilder content = new StringBuilder();
+        String line;
+
+        while ((line = reader.readLine()) != null) {
+            content.append(line);
+            content.append(System.lineSeparator());
+        }
+
+        return content.toString();
+    }
+
+    public String readAllLines2(BufferedReader reader) {
+        StringBuilder content = new StringBuilder();
+        reader.lines()
+            .forEach(line -> content.append(line)
+                .append(System.lineSeparator()));
+        return content.toString();
+    }
+
+    public String readAllCharacters(BufferedReader reader) throws IOException {
+        StringBuilder content = new StringBuilder();
+
+        int value;
+        while ((value = reader.read()) != -1) {
+            content.append((char) value);
+        }
+
+        return content.toString();
+    }
+
+    public String readAllCharacters2(BufferedReader reader) throws IOException {
+        StringBuilder content = new StringBuilder();
+
+        String title = "text: ";
```

Comment:
I'd be wary of adding additional logic in this method. The reason is that folks are going to want to compare `readAllCharacters` to `readAllCharacters2` and folks will probably see things like a title and line separators as noise.